### ISSUE TITLE
Create a training, validation and test set for note/seg

### ISSUE THREAD

**Issue by MansMeg** [welfare-state-analytics/riksdagen-corpus#189]
Current status (2024-01-25):
- [x] A bunch of training data
- [x] Proper train set
- [x] Validation set
- [x] Test set
- [x] BERT implementation

We probably want to have the student pool create the train set, and the Riksdagen people create the validation set. Then we also need to cross-check their work by having them annotate part of the test set too.

**Comment by ninpnin** [comment]
The code is here: https://github.com/welfare-state-analytics/riksdagen-corpus/tree/main/input/multi_label_classifier , we still need to run it (and check for errors)

**Comment by ninpnin** [comment]
We should revive this. IIRC the code works mostly out of the box.

- [ ] Annotate data, starting with maybe 10-30 pages per decade. Include data from the 1867-1920 period.
- [ ] Create a train set from the corpus
    - I would include a couple paragraphs before the to-be-classified paragraph. Sometimes it is helpful to see what text came before to classify a subsequent paragraph. It's also easy to check whether this actually helps.
- [ ] Implement code to find most uncertain paragraphs
- [ ] Train model
- [ ] Annotate 50-100 most uncertain paragraphs 1-3 times

**Comment by Stubbendorff** [comment]
Notes in the margins, specifying the topic of the debate, are often merged with speech segments. Example from prot-1943--fk--7.xml:   <seg xml:id="i-8sb6ofs7VgmJ5JiFppb1xo">
              Om utredning ang. kvinnans fulla likställighet med mannen m.
              m. (Forts.) och baroner, biskopar och professorer i kammaren var
              man på vissa områden mera radikal än nu. |
            </seg> Many cases follow a similar pattern, a headline followed by "m m" and/or "(Forts.)". Also from the 1940s: statsbidrag till daghem och lekskolor m m
till underhåll av enskilda vägar m m
till vissa broförbindelser över lule älv m
till prisrabattering å vissa livsmedel m m
för hotell restauranger och kaféer m m

**Comment by Stubbendorff** [comment]
This case from prot-197677--27.xml is even trickier (marginal notes in bold): uppe i 154 240 000 kr. För de åtta första månderna 1976 uppgår
              importen **— Nr 27** till ett värde av 132859 000 kr. Motsvarande
              siffra för 1975 var ca **Tisdagen den** 82 345 000 kr. Således en
              import för ca 10,3 milj. kr. per månad de första **16 november 1976**
              åtta månaderna 1975 mot ca 16,6 milj. kr. per månad de åtta första
              månderna 1976.

**Comment by MansMeg** [comment]
This is a great example.

**Comment by fredrik1984** [comment]
maybe relate to #250 and #188

**Comment by ninpnin** [comment]
Current status (2023-12-13):
- [x] A bunch of training data
- [ ] Validation set
- [x] Test set
- [x] BERT implementation

We probably want to have the student pool create the validation set. Then we also need to cross-check their work by having them annotate part of the test set too.

**Comment by ninpnin** [comment]
Validation set done 👍
